.github/workflows/delete-deployment.yml

```diff
@@ -0,0 +1,36 @@
+name: Delete deployment
+run-name: ${{ github.event.pull_request.title || format('Delete deployment [{0}]', github.head_ref || github.ref_name) }}
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [closed]
+
+permissions:
+  id-token: write
+  contents: read
+
+concurrency: deploy-development-${{ github.head_ref || github.ref_name }}
+
+jobs:
+  delete-stack:
+    name: Delete stack
+    runs-on: ubuntu-latest
+    environment: development
+
+    steps:
+      - name: Get stack name
+        uses: govuk-one-login/github-actions/beautify-branch-name@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        id: get-stack-name
+        with:
+          usage: Stack name
+          prefix: preview-check-hmrc-api
+          length-limit: 128
+          verbose: false
+
+      - name: Delete stack
+        uses: govuk-one-login/github-actions/sam/delete-stacks@e6b6ed890b35904e1be79f7f35ffec983fa4d9db
+        with:
+          stack-names: ${{ steps.get-stack-name.outputs.pretty-branch-name }}
+          aws-role-arn: ${{ secrets.DEPLOYMENT_ROLE_ARN }}
```

Comment:
ARNs are not secret. These should be reverted back to variables like in other repos.